### ISSUE TITLE
Search: No focus state on search input field in modal.

### ISSUE THREAD

**Issue by katiebethbrown** [Automattic/jetpack#23398]
### Impacted plugin
Jetpack

### Steps to Reproduce

There is no visible focus state when user is focused on the modal search box.

<img width="277" alt="CleanShot 2022-03-11 at 14 40 53@2x" src="https://user-images.githubusercontent.com/42279181/157974302-5ab3c889-4bcc-43f2-b035-b7b981b60066.png">

### Other Information
This issue was surfaced to team 51 by an a11y contractor working on a partner site.

**Comment by annacmc** [comment]
The current focus state for the search input field is a yellow highlight, but only when there is text present. I can quite easily update this to have the same blue glow outline on focus, but first, I'd like a second opinion on how it looks. @a8ck3n  @kangzj @bluefuton what do you think here? I feel it's kind of intrusive on the search experience. But on the other hand, it is best practice so 🤷 open to ideas! 

![image](https://user-images.githubusercontent.com/30754158/181145883-ddd3b8a3-0a40-43ca-9bb1-4bb33d2c5639.png)

**Comment by a8ck3n** [comment]
Yes, I agree it looks a bit odd like that. You can fix it by adding some padding around the input field though. For example…

![InputWithPadding](https://user-images.githubusercontent.com/40267301/181193178-f28da081-361c-40aa-9db4-b96ca823c98d.png)

That's with `padding: 0px 10px;` applied to the input element.

**Comment by annacmc** [comment]
> Yes, I agree it looks a bit odd like that. You can fix it by adding some padding around the input field though. For example…
> 
> ![InputWithPadding](https://user-images.githubusercontent.com/40267301/181193178-f28da081-361c-40aa-9db4-b96ca823c98d.png)
> 
> That's with `padding: 0px 10px;` applied to the input element.

I agree the padding helps, but I still wonder if it's quite intrusive. I do know that the focus state for this element was intentionally removed by Jason about 18 months ago. I would imagine for aesthetic reasons. I will create a PR though and we can see what everyone thinks.

**Comment by bluefuton** [comment]
@annacmc potentially we could show the focus outline for keyboard users, but not for mouse users:

https://css-tricks.com/the-focus-visible-trick/

**Comment by annacmc** [comment]
> @annacmc potentially we could show the focus outline for keyboard users, but not for mouse users:
> 
> [css-tricks.com/the-focus-visible-trick](https://css-tricks.com/the-focus-visible-trick/)

Thanks, @bluefuton that is a cool trick! I hadn't discovered that one before! After a lot of messing around with it, it turns out this won't work the same on input fields as they behave differently from other elements. 

focus indicators are displayed on text fields any time they’re focused, whether by keyboard or mouse, because it's expected that keyboard input will affect their contents. So the `&:focus:not( :focus-visible )` will work on other elements (eg buttons) but not on a form input like our search input. 

So sadly, this won't work here, and we are going to end up with the focus indicator displaying on the field whenever it's focused. We could perhaps make the focus different from the other elements, eg paler or thinner or something?

**Comment by a8ck3n** [comment]
> We could perhaps make the focus different from the other elements, eg paler or thinner or something?

Personally, I'd be in favour of using the default focus settings. In that way, the user can determine the colour and it should properly match the system. From my screenshot above, you can see my highlight colour is pink vs Anna's blue. I understand the desire to remove focus rings from a design perspective when the look isn't what you are going for. That being said, we should probably favour accessibility over design preferences at the end of the day.

**Comment by annacmc** [comment]
> > We could perhaps make the focus different from the other elements, eg paler or thinner or something?
> 
> Personally, I'd be in favour of using the default focus settings. In that way, the user can determine the colour and it should properly match the system. From my screenshot above, you can see my highlight colour is pink vs Anna's blue. I understand the desire to remove focus rings from a design perspective when the look isn't what you are going for. That being said, we should probably favour accessibility over design preferences at the end of the day.

I agree with you here, Kevin. Favouring accessibility seems to be a good choice! I was a little hesitant because of the work put in to actively suppress the focus ring, in this case, 😂  but let's try the system default highlight plus padding and see how we go. Putting together a PR for that now.

**Comment by bluefuton** [comment]
Sounds like a plan (and thanks for digging into focus-visible a bit more). We can ask for design review on your PR 👍
